### PR TITLE
smooth out progress bar

### PR DESCRIPTION
**Pull request by ibeckermayer**
Adds an interval for smoothing out the progress bar.

### CODE REVIEW

File: packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx [hunk 1]
```diff
@@ -34,54 +35,94 @@ export const ProgressBarDesktop = (props: {
...
-      playerClient.addListener(TdpClientEvent.TDP_ERROR, () => {
-        progressToEnd();
+      playerClient.addListener(TdpClientEvent.WS_CLOSE, () => {
+        stopProgress();
       });
 
       return () => {
         throttledUpdateCurrentTime.cancel();
         playerClient.nuke();
+        stopProgress();
```
Comment (kimlisa) [inline packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx:125]: line 123 no longer needed?

File: packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx [hunk 2]
```diff
@@ -34,54 +35,94 @@ export const ProgressBarDesktop = (props: {
...
+                current: nextTimeMs,
+                time: toHuman(nextTimeMs),
+              };
+            } else {
+              stopProgress();
+              return {
+                ...prevState,
+                current: durationMs,
+                time: toHuman(durationMs),
+              };
```
Comment (kimlisa) [inline packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx:70]: can we re-use `updateCurrentTime` here?

**Review by kimlisa — APPROVED** [review]
after addressing minor comments

File: packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx (hunk 1 above)
Comment (ibeckermayer) [reply packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx:125]: It's still needed, that function is still called when we get events from the server (carrying TDP messages)

**Review by hatched — APPROVED** [review]

File: packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx (hunk 1 above)
Comment (kimlisa) [reply packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx:125]: even when stopProgress calls it already? :thinking:

Comment (ibeckermayer) [reply packages/teleport/src/Player/ProgressBar/ProgressBarDesktop.tsx:121]: Oh no, I forgot that I'd added that. Thanks.

**Pull request merged**